api/src/main/java/com/basistech/rosette/api/RosetteAPI.java

```diff
@@ -166,6 +172,42 @@ public RosetteAPI(String key, String alternateUrl) throws IOException, RosetteAP
     }
 
     /**
+     * Constructs a Rosette API instance using the builder syntax.
+     *
+     * @param key            Rosette API key
+     * @param alternateUrl   Alternate Rosette API URL
+     * @param failureRetries Number of times to retry in case of failure; default 1
+     * @param language       source language
+     * @param genre          genre (ex "social-media")
+     * @param options        options for the request
+     * @throws IOException          General IO exception
+     * @throws RosetteAPIException  Problem with the API request
+     */
+    private RosetteAPI(String key, String alternateUrl, int failureRetries,
+                       LanguageCode language, String genre, Options options,
+                       CloseableHttpClient httpClient)
+                        throws IOException, RosetteAPIException {
+        Objects.requireNonNull(alternateUrl, "alternateUrl cannot be null");
+        this.key = key;
+        this.language = language;
+        this.genre = genre;
+        this.options = options;
+        urlBase = alternateUrl;
+        if (urlBase.endsWith("/")) {
+            urlBase = urlBase.substring(0, urlBase.length() - 1);
```

Comment:
would `urlBase.trim().replaceAll("/+$", "")` be better if there are accidental trailing spaces and multiple slashes?